### PR TITLE
Validate links

### PR DESCRIPTION
**Pull request by Ray725**

### CODE REVIEW

File: tools/validators/instance_validator/validate/entity_instance.py [hunk 1]
```diff
@@ -142,4 +170,16 @@ def IsValidEntityInstance(self):
         print('Missing required key:', req_key)
         return False
 
-    return self._ValidateType() and self._ValidateTranslation()
+    if not self._ValidateType():
+      print('Invalid type key')
```
Comment (charbull) [inline tools/validators/instance_validator/validate/entity_instance.py:174]: print the key value

File: tools/validators/instance_validator/validate/entity_instance.py [hunk 2]
```diff
@@ -142,4 +170,16 @@ def IsValidEntityInstance(self):
...
+      return False
+
+    if not self._ValidateLinks():
+      print('Invalid links key')
```
Comment (charbull) [inline tools/validators/instance_validator/validate/entity_instance.py:178]: print the key value

File: tools/validators/instance_validator/validate/entity_instance.py [hunk 3]
```diff
@@ -142,4 +170,16 @@ def IsValidEntityInstance(self):
...
+      return False
+
+    if not self._ValidateTranslation():
+      print('Invalid translation key')
```
Comment (charbull) [inline tools/validators/instance_validator/validate/entity_instance.py:182]: print the key value

File: tools/validators/instance_validator/validate/entity_instance.py [hunk 4]
```diff
@@ -67,6 +69,32 @@ def _ValidateType(self):
...
+      key is not present.
+    """
+    if self.links not in self.entity.keys():
+      return True
+
+    links = dict(self.entity[self.links])
+    for entity_name in links.keys():
+      # ensure first level keys refer to other entities in config file
+      if entity_name not in self.config_entity_names:
+        return False
```
Comment (charbull) [inline tools/validators/instance_validator/validate/entity_instance.py:87]: error message?

File: tools/validators/instance_validator/validate/entity_instance.py [hunk 5]
```diff
@@ -67,6 +69,32 @@ def _ValidateType(self):
...
+
+      # scan all standard fields and ensure they're defined
+      fields_map = dict(links[entity_name])
+      field_universe = self.universe.field_universe
+      for sourcename in fields_map.keys():
+        if not field_universe.IsFieldDefined(fields_map[sourcename], ''):
```
Comment (charbull) [inline tools/validators/instance_validator/validate/entity_instance.py:93]: Error message

File: tools/validators/instance_validator/validate/entity_instance.py [hunk 6]
```diff
@@ -27,11 +27,13 @@ class EntityInstance(findings_lib.Findings):
     universe: ConfigUniverse generated from the ontology
   """
```
Comment (charbull) [inline tools/validators/instance_validator/validate/entity_instance.py:28]: update doc with the new arg

File: tools/validators/instance_validator/validate/entity_instance.py (hunk 1 above)
Comment (Ray725) [reply tools/validators/instance_validator/validate/entity_instance.py:174]: Done, added in `_ValidateTranslation`

File: tools/validators/instance_validator/validate/entity_instance.py (hunk 2 above)
Comment (Ray725) [reply tools/validators/instance_validator/validate/entity_instance.py:178]: Done, added in `_ValidateLinks()`

File: tools/validators/instance_validator/validate/entity_instance.py (hunk 3 above)
Comment (Ray725) [reply tools/validators/instance_validator/validate/entity_instance.py:182]: Done, added in _ValidateTranslation

File: tools/validators/instance_validator/validate/entity_instance.py (hunk 1 above)
Comment (Ray725) [reply tools/validators/instance_validator/validate/entity_instance.py:174]: *already exists in `_ValidateType()`

File: tools/validators/instance_validator/validate/entity_instance.py (hunk 4 above)
Comment (Ray725) [reply tools/validators/instance_validator/validate/entity_instance.py:87]: Done

File: tools/validators/instance_validator/validate/entity_instance.py (hunk 5 above)
Comment (Ray725) [reply tools/validators/instance_validator/validate/entity_instance.py:93]: Done

File: tools/validators/instance_validator/validate/entity_instance.py (hunk 6 above)
Comment (Ray725) [reply tools/validators/instance_validator/validate/entity_instance.py:28]: Done

**Pull request merged**